Fix istft default output length

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80031

Fixes #79778

The old code uses `end = -1` for non-centered input when no length is
given, however `end` should point one past the end, so it was trimming
away the last element.

peterbell10 committed:
Fix istft default output length

The old code uses `end = -1` for non-centered input when no length is
given, however `end` should point one past the end, so it was trimming
away the last element.

[ghstack-poisoned]

mruberry reviewed (approved):
Stamped!

Once the jobs are green you can merge this using pytorchbot, @peterbell10

peterbell10 committed:
Rebase on "Fix istft default output length"


Fixes #79778

The old code uses `end = -1` for non-centered input when no length is
given, however `end` should point one past the end, so it was trimming
away the last element.

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge -g
